perf/perf_nest_events.py

```diff
@@ -0,0 +1,111 @@
+#
+# See LICENSE for more details.
+#
+# Copyright: 2019 IBM
+# Author: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>
+
+import os
+import platform
+import tempfile
+from avocado import Test
+from avocado import main
+from avocado.utils import cpu, distro, genio, process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class nestEvents(Test):
+
+    """
+    Tests nest events
+    Collects all the available events from 'perf list' and
+    executes them using 'perf stat' command
+    """
+    # Initializing fail command list
+    fail_cmd = list()
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.distro_name = detected_distro.name
+        self.cpu_arch = cpu.get_cpu_arch().lower()
+
+        if detected_distro.arch != 'ppc64le':
+            self.cancel('This test is not supported on %s architecture'
+                        % detected_distro.arch)
+        if self.cpu_arch == 'power8':
+            self.cancel('This test not applies to Power8')
+        if 'PowerNV' not in genio.read_file('/proc/cpuinfo').rstrip('\t\r\n\0'):
+            self.cancel('This test applies only to PowerNV')
+
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in self.distro_name:
+            deps.extend(['linux-tools-common', 'linux-tools-%s' %
+                         platform.uname()[2]])
+        elif self.distro_name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(['perf'])
+        else:
+            self.cancel("Install the package for perf supported \
+                         by %s" % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        # Create the file with contents of nest events
+        self.perf_list_file = tempfile.NamedTemporaryFile().name
+        output = process.run("perf list")
+        for line in output.stdout.split('\n'):
+            if 'nest_' in line:
+                line = line.split(' ')[2]
+                if 'pm_nest' in line:
+                    continue
+                genio.append_one_line(self.perf_list_file, line)
+
+        # Clear the dmesg, by that we can capture the delta at the end of the test.
+        output = process.run("dmesg -c")
+
+    def error_check(self):
+        if len(self.fail_cmd) > 0:
+            for cmd in range(len(self.fail_cmd)):
+                self.log.info("Failed command: %s" % self.fail_cmd[cmd])
+            self.fail("perf_raw_events: some of the events failed, refer to log")
+
+    def run_cmd(self, cmd):
+        output = process.run(cmd, shell=True,
+                             ignore_status=True)
+        if output.exit_status != 0:
+            self.fail_cmd.append(cmd)
```

Comment:
i think output.stdout you are interested  so it can be like 

```
if process.system(cmd, shell=True, ignore_status=True):
   self.fail_cmd.append(cmd)
```

it is to avoid extra variable